pcdsutils/log.py

```diff
@@ -461,132 +461,252 @@ def uninstall_log_warning_handler() -> None:
+            if info in self.cache or not self.only_duplicates:
+                record.levelno = self.levelno
+                record.levelname = self.levelname
+                self.counter += 1
+            else:
+                self.cache.add(info)
+        return True
+
+    def reset_counter(self):
+        self.counter = 0
+
+    def should_demote(
+        self,
+        record: logging.LogRecord,
+        info: RecordInfo
+    ) -> bool:
+        """
+        Return True if we should demote the record, False otherwise.
+
+        Parameters
+        ----------
+        record: logging.LogRecord
+            The record instance passed into the filter.
+        info: RecordInfo
+            The record info dataclass, which may be more convenient
+            for implementing this method.
+        """
+        raise NotImplementedError("Implement me in child class")
+
+
+@dataclasses.dataclass(eq=True, frozen=True)
+class WarningRecordInfo(RecordInfo):
+    """
+    Hashable collection of the unique information from a warnings.warn call.
+    """
+    category: type[Warning]
+    filename: str
+    lineno: int
+
+    @classmethod
+    def from_record(cls, record: logging.LogRecord) -> WarningRecordInfo:
+        """
+        Create a WarningRecordInfo from a LogRecord.
+
+        This can be used as a utility to inspect or compare warnings log
+        messages inside a log filter.
+        """
+        try:
+            return cls(
+                message=str(record.warning_message),
+                category=record.warning_category,
+                filename=record.warning_filename,
+                lineno=record.warning_lineno,
+            )
+        except AttributeError as exc:
+            raise ValueError(
+                'Received invalid record, must be from '
+                'the log_warning_handler'
+            ) from exc
+
+
+class LogWarningLevelFilter(DemotionFilter):
+    """
+    Filter to decrease the log level of repeat warnings.
+
+    Once installed, the first instance of a "warnings.warn" converted to
+    a log message will be at "WARNING" level, while subsequent repeats of
+    the same warning will be at "DEBUG" level.
+
+    When running a normal program, typically the warnings module will
+    handle warning filtering for you as the default behavior, making every
+    unique warning only appear once.
+
+    When running in ipython, the warnings cache is reset prior to running
+    every command, so it's possible to see repeat warnings during or after
+    each command. This is a annoying. The filter here allows us to
+    adjust the level of the repeat messages to avoid cluttering the user's
+    view, or to remove them entirely.
+
+    This filter incorporates a resettable counter that will count the number
+    of demoted log messages.
+
+    Parameters
+    ----------
+    level : str or int, optional
+        The log level or name of the log level to reduce dupliacte
```

Comment:
```suggestion
        The log level or name of the log level to reduce duplicate
```

Reply:
phew, I only copy/pasted this typo in two places